core/src/main/scala/io/delta/tables/DeltaTableBuilder.scala

```diff
@@ -109,7 +109,8 @@ class DeltaTableBuilder private[tables](
   private var columns: mutable.Seq[StructField] = mutable.Seq.empty
   private var location: Option[String] = None
   private var tblComment: Option[String] = None
-  private var properties = CaseInsensitiveMap[String](Map.empty)
```

Comment:
Do you know why `delta.appendOnly` is not affected by this issue?

Reply:
The problem seems to come from the fact that when the metadata is read from the snasphot, properties are merged with default properties. I checked in the logical and physical plan as well and the properties were all in lowercase.

https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L187

This code relies on a pre-initialized map called "entries" which you see here https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L133